repos/system_upgrade/el7toel8/libraries/testutils.py

```diff
@@ -24,6 +28,53 @@ def __call__(self, report_fields):
             self.report_fields.update(report.to_dict())
 
 
+# class run_mocked(object):
+
+class logger_mocked(object):
+    def __init__(self):
+        self.dbgmsg = []
+        self.infomsg = []
+        self.warnmsg = []
+        self.errmsg = []
+
+    def debug(self, *args):
+        self.dbgmsg.extend(args)
+
+    def info(self, *args):
+        self.infomsg.extend(args)
+
+    def warn(self, *args):
+        self.warnmsg.extend(args)
+
+    def warning(self, *args):
+        self.warnmsg.extend(args)
+
+    def error(self, *args):
+        self.errmsg.extend(args)
+
+    def __call__(self):
+        return self
+
+
+class CurrentActorMocked(object):
+    def __init__(self, arch=architecture.ARCH_X86_64, envars=None, kernel='3.10.0-957.43.1.el7.x86_64',
+                 release_id='rhel', src_ver='7.8', dst_ver='8.1'):
+        envarsList = [EnvVar(name=k, value=v) for k, v in envars.items()] if envars else []
+        version = namedtuple('Version', ['source', 'target'])(src_ver, dst_ver)
+        release = namedtuple('OS_release', ['release_id', 'version_id'])(release_id, src_ver)
+
+        self._common_folder = '../../files'
+        self.configuration = namedtuple(
+            'configuration', ['architecture', 'kernel', 'leapp_env_vars', 'os_release', 'version']
+        )(arch, kernel, envarsList, release, version)
+
+    def __call__(self):
+        return self
+
+    def get_common_folder_path(self, folder):
+        return os.path.join(self._common_folder, folder)
+
+
```

Comment:
For the PR #494 and #493 I need a mocked consume method and log property implemented. Something like this:
```python
class CurrentActorMocked(object):
    def __init__(
        self,
        arch=architecture.ARCH_X86_64,
        envars=None,
        kernel='3.10.0-957.43.1.el7.x86_64',
        release_id='rhel',
        src_ver='7.8',
        dst_ver='8.1',
        msgs=None,
    ):
        envarsList = (
            [EnvVar(name=k, value=v) for k, v in envars.items()]
            if envars
            else []
        )
        version = namedtuple('Version', ['source', 'target'])(src_ver, dst_ver)
        release = namedtuple('OS_release', ['release_id', 'version_id'])(
            release_id, src_ver
        )

        self._common_folder = '../../files'
        self.configuration = namedtuple(
            'configuration',
            [
                'architecture',
                'kernel',
                'leapp_env_vars',
                'os_release',
                'version',
            ],
        )(arch, kernel, envarsList, release, version)
        self._msgs = [] if not msgs else msgs

    def __call__(self):
        return self

    def get_common_folder_path(self, folder):
        return os.path.join(self._common_folder, folder)

    def consume(self, model):
        return filter(  # pylint:disable=W0110,W1639
            lambda msg: isinstance(msg, model), self._msgs
        )

    @property
    def log(self):
        return logging.getLogger(LEAPP_LOGGER_NAME)
```

Reply:
@drehak what about this implementation of the log property?